### ISSUE TITLE
AnimatedTimeBasedSprite's update routine very slow.

### ISSUE THREAD

**Issue by Schiggebam** [pvcraven/arcade#684]
Hello!
This is not so much a bug report, but something I have noticed while playing around with the engine:
I am using a simple class which inherits from arcade.AnimatedTimeBasedSprite, which loades a spritesheed like so:
```
class Flag(arcade.AnimatedTimeBasedSprite):
    def __init__(self, x, y):
        super().__init__()
        path = "../resources/objects/animated/flag_100_sprite_green.png"
        for i in range(10):
            tex: arcade.Texture = arcade.load_texture(path, x=0, y=i*100, width=108, height=100)
            self.append_texture(tex)
            a = AnimationKeyframe(i, 50, tex)
            self.frames.append(a)
        self.update_animation()
```
I put them all in a spritelist and in the update thread I call:
`self.flag_sl.update_animation()`
When I use about ~50 of these "flags" at once, the FPS drop already below 15 on a decently modern machine. 
My question is: Do I do something drastically wrong, or is this behaviour expected, that some 50+ animated sprites push the engine to its limits?

Thank you very much for an answer!

**Comment by pvcraven** [comment]
It should work ok. The explosion example can flip through a lot of textures.

I'd recommend coming up with a self contained example, zipping it up and attaching to this report. Or run a profiler and narrow down where the time is being spent.

**Comment by Schiggebam** [comment]
Thank you very much for your quick answer.
Please find a minimal example, derived from an example code on your website below.


![flag_100_sprite_green](https://user-images.githubusercontent.com/4637628/81088797-c9b15900-8efb-11ea-8c77-b60a5f4479cf.png)
[example_1.py.txt](https://github.com/pvcraven/arcade/files/4582082/example_1.py.txt)

**Comment by Schiggebam** [comment]
Another way of "reproducing this issue" is if I use your explosion example and I change one line of code in the update method of the Explosion class:
```
    def update(self):

        # Update to the next frame of the animation. If we are at the end
        # of our frames, then delete this sprite.
        self.current_texture += 1
        if self.current_texture < len(self.textures):
            self.set_texture(self.current_texture)
        else:
            self.current_texture = 0
            #self.remove_from_sprite_lists()
```
This will keep the explosions "alive" and running. 
Now, once ~30 explosions are triggered, the update routine slows down notably.

To me, this comes as a surprise. I feel like flipping through "static" sprite_lists shouldn't be very costly, where everything can be held/managed in the GPU

**Comment by pvcraven** [comment]
Thanks, perfect example.

**Comment by pvcraven** [comment]
Version 2.4a6 sometimes gives me update animation times of 0.211 seconds, and it settles on 12 FPS.

Version 2.4a7 gives my worst update animation times of 0.00125 seconds, and settles on a FPS of about 59 FPS once it gets going.

Please try 2.4a7 and see if it works better for you.

**Comment by Schiggebam** [comment]
Once again, thank you very much for your answer! Before, I was using acrade 2.3.15

Amazingly, the set_texture(int) routine appears to be of a factor 1000 faster than in the version before! This definitely solved the issue.

In my first example, the flags stop "waving" after one cycle, but I will figure it out.

All in all, this greatly improved the performance of the simple game I was hacking together, also the draw time without any keyframe animations is cut in half :-)

I will have a look at your code the see what was changed, thanks again! Great project!

**Comment by pvcraven** [comment]
Ah! Sorry, forgot to mention. is_static assumes sprites won't change. You MUST turn that off to get more than one cycle.

**Comment by pvcraven** [comment]
Released in 2.4.
